rpcs3/rpcs3qt/user_manager_dialog.cpp

```diff
@@ -0,0 +1,277 @@
+	connect(push_create_user, &QAbstractButton::clicked, this, &user_manager_dialog::OnUserCreate);
+	connect(push_login_user, &QAbstractButton::clicked, this, &user_manager_dialog::OnUserLogin);
+	connect(m_table, &QTableWidget::itemDoubleClicked, this, &user_manager_dialog::OnUserLogin);
+	connect(m_table->horizontalHeader(), &QHeaderView::sectionClicked, this, &user_manager_dialog::OnSort);
+	connect(m_table, &QTableWidget::customContextMenuRequested, this, &user_manager_dialog::ShowContextMenu);
+	connect(m_table, &QTableWidget::cellChanged, [&](int row, int col)
+	{
+		int originalIndex = m_table->item(row, 0)->data(Qt::UserRole).toInt();
+		UserAccount* originalEntry = m_user_list[originalIndex];
+		QString originalDirName = qstr(originalEntry->GetUserDir());
+	});
+}
+
+void user_manager_dialog::UpdateTable()
+{
+	if (m_dir == "")
+	{
+		// fmt::format(%shome ... is harder to read than straight concatenation.
+		m_dir = Emu.GetHddDir() + "home";
+	}
+
+	// Get the user folders in the home directory.
+	m_user_list = GetUserAccounts(m_dir);
+
+	// Clear and then repopulate the table with the list gathered above.
+	m_table->clearContents();
+	m_table->setRowCount(static_cast<int>(m_user_list.size()));
+
+	int row = 0;
+	for (UserAccount* user : m_user_list)
+	{
+		// For indicating logged-in user.
+		QFont boldFont;
+		boldFont.setBold(true);
+
+		QString userId = qstr(user->GetUserId());
+		QString userName = qstr(user->GetUserName());
+
+		QTableWidgetItem* userIdItem = new QTableWidgetItem(userId);
+		userIdItem->setData(Qt::UserRole, row); // For sorting to work properly
+		userIdItem->setFlags(userIdItem->flags() & ~Qt::ItemIsEditable);
+
+		// Compare current config value with the one in this user (only 8 digits in userId)
+		if (m_emu_settings->GetSetting(emu_settings::SelectedUser)
+			.compare(0, 8, user->GetUserId()) == 0)
+		{
+			userIdItem->setFont(boldFont);
+		}
+		m_table->setItem(row, 0, userIdItem);
+
+		QTableWidgetItem* userNameItem = new QTableWidgetItem(userName);
+		userNameItem->setData(Qt::UserRole, row); // For sorting to work properly
+		userNameItem->setFlags(userNameItem->flags() & ~Qt::ItemIsEditable);
+
+		// Compare current config value with the one in this user (only 8 digits in userId)
+		if (m_emu_settings->GetSetting(emu_settings::SelectedUser)
+			.compare(0, 8, user->GetUserId()) == 0)
+		{
+			userNameItem->setFont(boldFont);
+		}
+		m_table->setItem(row, 1, userNameItem);
+
+		++row;
+	}
+
+	// GUI resizing
+	m_table->horizontalHeader()->resizeSections(QHeaderView::ResizeToContents);
+	m_table->verticalHeader()->resizeSections(QHeaderView::ResizeToContents);
+
+	QSize tableSize = QSize(
+		m_table->verticalHeader()->width() + m_table->horizontalHeader()->length() + m_table->frameWidth() * 2,
+		m_table->horizontalHeader()->height() + m_table->verticalHeader()->length() + m_table->frameWidth() * 2);
+
+	QSize preferredSize = minimumSize().expandedTo(sizeHint() - m_table->sizeHint() + tableSize);
```

Comment:
add `.expandedTo(size());` in order to fix unwanted size reset

Reply:
Ah, that explains that.